src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.maxRotations = maxRotations;
+    this.mainGearCircumference = mainGearCircumference;
+  }
+
+  //"These two functions speak for themselves" - Hans Niemann commenting his code
+  public static double gcd(double a, double b) {
+    if (b == 0.0)
+      return a;
+    return gcd(b, a % b);
+  }
+
+  public static double mod(double a, double b) {
+    double r = a % b;
+    if (r < 0) {
+      r += b;
+    }
+    return r;
+  }
+
+  /**
+   * 
+   * @return Returns a multiplier used by CRT. You shouldn't need to use this function yourself, and
+   *         I don't fully understand it (Victor Chen did the math behind it).
+   */
+  public int kFactor() {
+    double sum = 0;
+    double a = secondaryGearRotationRatioTermSimplfied;
+    double b = mainGearRotationRatioTermSimplified;
+    for (int i = 1; i < b; i++) {
+      sum += ((a * (b - i) - 1.0) / b)
+          * (1 + Math.floor(((a * (b - i) - 1.0) / b)) - Math.ceil(((a * (b - i) - 1.0) / b)));
+    }
+    return (int) ((int) ((sum * b) + 1.0) / a); //basically, when you apply CRT, you can multiply by a certain factor (what this function returns, a constant for a given gear ratio) such that when using modding that factor times the output of crt and the simplified main gear rotation ratio term, it will correctly output a number between 0 and maxrotations. then, to account for wrap around and to make 0 full rotations work correctly, mod the main gear ratio term by the output + the simplified term
+  }
+
+  /**
+   * 
+   * @param mainGearRotationValue      Rotation value returned by the main gear
+   *                                   encoder
+   * @param secondaryGearRotationValue Rotation value returned by the secondary
+   *                                   gear encoder
+   * @return Returns the number of full rotations of the main gear
+   */
+  public int getMainGearFullRotations(double mainGearRotationValue, double secondaryGearRotationValue) {
+    double k = kFactor();
+    return (int) Math.round(mod(mainGearRotationRatioTerm, //mod the main gear rotation ratio term by the whole thing plus the simplified term to make 0 full rotations work properly
+        mod(k * //this is the multiplier, see the comment in the returns statement of kFactor() for more details
+        (mainGearRotationValue * secondaryGearRotationRatioTermSimplfied
+            - secondaryGearRotationValue * mainGearRotationRatioTermSimplified) //this returns a decimal where the multiplier is different for every gear ratio
+            , mainGearRotationRatioTermSimplified) //mod it with the main gear rotation ratio term so it is within the correct range
+            + mainGearRotationRatioTermSimplified)); //for more information, see the comment in the kFactor() function
+  }
+
+  /**
+   * 
+   * @return Returns true if the gear ratio works with the neccessary number of
+   *         max rotations, false otherwise
```

Comment:
Here max rotations is used as the desired max

Reply:
fixed